docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -12,35 +13,22 @@ Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
 
 ## Decision
 
-When using the `oneof` protobuf type there are many times where one will have to manually switch over the possible messages and then pass them to the interface which is needed. By transitioning from a fixed size byte array (`[size]byte`) to byte slice's (`[]byte`) then this would enable the usage of the [cosmos-proto's](hhttps://github.com/regen-network/cosmos-proto#interface_type) interface type, which will generate these switch statements.
+Previously Tendermint defined all the key types for use in Tendermint and the Cosmos-SDK. Going forward the Cosmos-SDK will define its own protobuf type for keys. This will allow Tendermint to only define the keys that are being used in the codebase (ed25519).
+There is the the opportunity to only define the usage of ed25519 (`bytes`) and not have it be a `oneof`, but this would mean that the `oneof` work is only being postponed to a later date. When using the `oneof` protobuf type we will have to manually switch over the possible key types and then pass them to the interface which is needed.
```

Comment:
are u saying we'll drop support for secp256k1 and sr25519, but later may add them back?

Reply:
no just that when we decide to add a curve later one we would just add it to the oneof instead of changing the code all over again. I dont see us adding secp back but adding other curves like bls